caikit_tgis_backend/tgis_backend.py

```diff
@@ -176,25 +168,52 @@ def get_connection(
         if not model_conn and create and not self.local_tgis and conn_cfg:
             model_conn = TGISConnection.from_config(model_id, conn_cfg)
             if self._test_connections:
-                try:
-                    model_conn.test_connection()
-                except grpc.RpcError as err:
-                    log.warning(
-                        "<TGB50048960W>",
-                        "Unable to connect to model %s: %s",
-                        model_id,
-                        err,
-                        exc_info=True,
-                    )
-                    model_conn = None
+                model_conn = self._test_connection(model_conn)
             if model_conn is not None:
-                # NOTE: setdefault used here to avoid the need to hold the mutex
-                #   when running the connection test. It's possible that two
-                #   threads would stimulate the creation of the connection
-                #   concurrently, so just keep whichever dict update lands first
-                self._model_connections.setdefault(model_id, model_conn)
+                self._safely_update_state(model_id, model_conn)
+
         return model_conn
 
+    def register_model_connection(
+        self,
+        model_id: str,
+        conn_cfg: Optional[Dict[str, Any]] = None,
+        fill_with_defaults: bool = True,
+    ) -> None:
+        """
+        Register a remote model connection.
+
+        If the model connection is already registered, do nothing.
+
+        Otherwise create and register the model connection using the TGISBackend's
+        config connection, or the `conn_cfg` if provided.
+
+        If `fill_with_defaults == True`, missing keys in `conn_cfg` will be populated
+        with defaults from the TGISBackend's config connection.
+        """
+        if model_id in self._model_connections:
+            return  # Model connection exists --> do nothing
+
+        # Craft new connection config
+        new_conn_cfg = {}
+        if conn_cfg is None:
+            new_conn_cfg = self._base_connection_cfg
+        else:
+            if fill_with_defaults:
+                new_conn_cfg = self._base_connection_cfg
```

Comment:
I think this needs to be `self._base_connection_cfg.copy` otherwise the `update` below will mutate `self._base_connection_cfg`. Since there is nesting, it should actually probably be `copy.deepcopy`.

Reply:
Good catch. fixed.